base/ca/src/org/dogtagpki/ct/CTEngine.java

```diff
@@ -0,0 +1,745 @@
+            if (extensions_s == null) {
+                extensions_s = "";
+            }
+            byte[] extensions = CryptoUtil.base64Decode(extensions_s);
+
+            // piece them together
+            int data_len = version.length + signature_type.length +
+                     timestamp.length + entry_type.length +
+                     issuer_key_hash.length
+                     + 3 + tbsCert.length
+                     + 2 + extensions.length;
+            logger.debug(method + " data_len = "+ data_len);
+
+            ByteArrayOutputStream ostream = new ByteArrayOutputStream();
+
+            ostream.write(version);
+            ostream.write(signature_type);
+            ostream.write(timestamp);
+
+            ostream.write(entry_type);
+            ostream.write(issuer_key_hash);
+
+            // 3 bytes for length of tbsCert
+            ostream.write(intToFixedWidthBytes(tbsCert.length, 3));
+            ostream.write(tbsCert);
+
+            // 2 bytes for extensions len
+            ostream.write(intToFixedWidthBytes(extensions.length, 2));
+            ostream.write(extensions);
+
+            byte[] data = ostream.toByteArray();
+            logger.debug(method + "actual data len = " + data.length);
+
+            Signature signer = Signature.getInstance(
+                    hashAlg + "with"+ sigAlg, "Mozilla-JSS");
+            signer.initVerify(log_pubKey);
+            signer.update(data);
+
+            return signer.verify(signature);
+        } catch (Throwable e) {
+            logger.debug(method + "Exception thrown: " + e.toString(), e);
+            return false;
+        } finally {
+            logger.debug(method + "ends");
+        }
+    }
+
+    /**
+     * parses and gleans the hashing algorithm from the returned
+     * CT signature
+     *
+     *  enum HashAlgorithm {
+     *      none, md5, sha1, sha224, sha256, sha384, sha512};
+     */
+    enum HashAlgorithm {none, MD5, SHA1, SHA224, SHA256, SHA384, SHA512};
+    public String getHashAlgFromSig(byte[] ct_signature) {
+
+        int hashingAlg = Byte.toUnsignedInt(ct_signature[0]);
+        if (hashingAlg != 4) // only SHA256 supported for v1
+            return null;
+
+        return HashAlgorithm.values()[hashingAlg].name();
+    }
+
+    /**
+     * parses and gleans the signature algorithm from the returned
+     * CT signature
+     *
+     * enum SignatureAlgorithm { anonymous, rsa, dsa, ecdsa}
+     */
+    enum SignatureAlgorithm {anonymous, RSA, DSA, EC};
+    public String getSigAlgFromSig(byte[] ct_signature) {
+
+        int signingAlg = Byte.toUnsignedInt(ct_signature[1]);
+        if (signingAlg < 1 || signingAlg > 3)
+            return null;
+
+        return SignatureAlgorithm.values()[signingAlg].name();
+    }
+
+    /**
+     * Write the int as a big-endian byte[] of fixed width (in bytes).
+     */
+    public static byte[] intToFixedWidthBytes(int n, int width) {
```

Comment:
Minor: If this is a reusable method to the util package. If specific to SCT, move alone with `verifySCT()`